packages/app/search/service.ts

```diff
@@ -0,0 +1,29 @@
+import compile from 'monquery'
+import type { Model } from '../models/types'
+import type { ErrorData } from '../declarations'
+import { searchModelsDb } from './db'
+import log from '../lib/log'
+
+/**this searvice takes lucene query string and converts it to mongoDB query*/
+
+export function searchwithMonquery(searchQuery) {
+    let search = compile(searchQuery)
+
+    return search
+}
+
+/*this function throws an error if there is a model users and workflows. since we do not want this models to be searched.*/
+export async function getModel(modelName: string): Promise<ErrorData<Model>> {
```

Comment:
I don't think we need this extra getModel here. 

I can see the potential need to not allow searching Users and Workflows but I think we should keep that permission based handling in the original "getModel" instead of in search. Happy to chat about that, but I think we can just remove this method in favor of using models/ getModel instead

Reply:
I think I can handle that through the API endpoint I'm making. We have a `userCanAccessModel` function that I completely forgot about when we were talking earlier.

Reply:
Ok that sounds good. I saw userCanAccessModel function and we should be able to use it for this logic. I will remove get model from here.